web_dark_mode/static/src/js/switch_item.esm.js

```diff
@@ -0,0 +1,52 @@
+/** @odoo-module **/
+
+import {browser} from "@web/core/browser/browser";
+import {registry} from "@web/core/registry";
+
+export function darkModeSwitchItem(env) {
+    return {
+        type: "switch",
+        id: "color_scheme.switch",
+        description: env._t("Dark Mode"),
+        callback: () => {
+            env.services.color_scheme.switchColorScheme();
+        },
+        isChecked: env.services.cookie.current.color_scheme === "dark",
+        sequence: 40,
+    };
+}
+
+export const colorSchemeService = {
+    dependencies: ["cookie", "orm", "ui", "user"],
+
+    async start(env, {cookie, orm, ui, user}) {
+        registry.category("user_menuitems").add("darkmode", darkModeSwitchItem);
+        if (!cookie.current.color_scheme) {
+            const data = await orm.webSearchRead(
+                "res.users",
+                [["id", "=", user.userId]],
+                ["dark_mode"]
+            );
+
+            const dark_mode = data.length && data.records[0].dark_mode;
+            cookie.setCookie("color_scheme", dark_mode ? "dark" : "light");
+        }
```

Comment:
window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', event => {
    cookie.setCookie("color_scheme", event.matches ? "dark" : "light");
    browser.location.reload();
})
const dark_mode = window.matchMedia('(prefers-color-scheme: dark)').matches;
cookie.setCookie("color_scheme", dark_mode ? "dark" : "light");

Reply:
If we are talking about work environments, it can happen the same as you activate the dark mode or disable it if the equipment you use is not unipersonal, but I'm not saying to remove the on or off button, sorry if I did not explain, it would only be interesting that the default is the one on the device

Reply:
As far as I can see it can only happen in device dependent mode that the cookie isn't set by python code. In that case the dark_mode flag of res.users is just the last value selected on any device. In that case using the system preferences might be the better default. I'll give it a try. :+1:

Reply:
Alright the JS uses the prefers-color-scheme as default now if you have activated the device dependent mode.